src/lib/pake/spake2/spake2.h

```diff
@@ -0,0 +1,201 @@
+*
+* The key confirmation step is omitted; it is assumed that further
+* uses of the shared secret will confirm the key. The shared secret is
+* equivalent to Hash(TT) in RFC 9382 so it is possible to implement
+* RFC 9382 conformant key confirmation if necessary.
+*/
+class BOTAN_PUBLIC_API(3, 7) SPAKE2_Parameters final {
+   public:
+      /**
+      * RFC 9382 compatible SPAKE2 configuration
+      *
+      * The shared secret is hashed with Argon2id M=131072,t=3,p=1 and with the
+      * context string as the AD
+      *
+      * If the group is not P-256, P-384, or P-521, then the group must support
+      * RFC 9380 hash to curve. Curves with a cofactor are not supported.
+      *
+      * If per_user_params is true, this uses the "quantum annoying" variant
+      * where N/M are the output of hash to curve; this requires an attacker
+      * with a quantum computer to perform a discrete logarithm calculator per
+      * PAKE, rather than just once for the default (fixed) SPAKE2 parameters.
+      * This variant is described in RFC 9382 section 5, however note that a
+      * different input is used to hash_to_curve which includes not just the
+      * user identifiers but also the context string.
+      *
+      * @param group the elliptic curve group to operate in
+      * @param shared_secret the shared secret (eg a password)
+      * @param a_identity the (optional) identity string of peer A
+      * @param b_identity the (optional) identity string of peer B
+      * @param context an optional context string (for example a protocol identifier)
+      * @param hash the hash function to use (SHA-512 highly recommended)
+      * @param per_user_params if true then per-user N/M are used
+      */
+      SPAKE2_Parameters(const EC_Group& group,
+                        std::string_view shared_secret,
+                        std::span<const uint8_t> a_identity = {},
+                        std::span<const uint8_t> b_identity = {},
+                        std::span<const uint8_t> context = {},
+                        std::string_view hash = "SHA-512",
+                        bool per_user_params = true);
+
+      /**
+      * RFC 9382 compatible SPAKE2 configuration
+      *
+      * If the group is not P-256, P-384, or P-521, then the group must support
+      * RFC 9380 hash to curve. Curves with a cofactor are not supported.
+      *
+      * If per_user_params is true, this uses the "quantum annoying" variant
+      * where N/M are the output of hash to curve; this requires an attacker
+      * with a quantum computer to perform a discrete logarithm calculator per
+      * PAKE, rather than just once for the default (fixed) SPAKE2 parameters.
+      *
+      * Here the shared secret a random scalar. It should have been generated
+      * using a memory hard function such as Argon2id.
+      *
+      * # ⚠️  Warning
+      *
+      * This interface exists primarily for testing, and is not safe for general use.
```

Comment:
It does allow this (or Argon2id with different parameters, etc), so there is utility outside of the testing scenario. However it seems like an obvious footgun already, and a `span` argument instead seems even more prone to misuse.

I suppose I can amend the comment to make it clear it is usable as long as the scalar is derived from the password using a MHF and with care to avoid bias, referring to the relevant RFC sections.